extension-workspace-references.md

```diff
@@ -0,0 +1,104 @@
+# workspace/xreferences extension to LSP
+
+The `workspace/xreferences` extension to the Language Server Protocol (LSP) enables a language server to export all of the references made from the workspace's code to its dependencies. Additionally, a new `textDocument/xdefinition` method performs the equivalent of `textDocument/definition` while returning metadata about the definition and _optionally_ a concrete location to it.
+
+Use case: clients of a language server can invoke `workspace/xreferences` in order to find references to dependencies. This information can then be stored in a database, which allows the caller to create a "global mapping" of symbols in dependencies to the workspaces they are used in (e.g. to see "how do other people use this symbol?"). A user would perform `textDocument/xdefinition` in order to locate the metadata about the symbol they are interested in and find its references in the database.
+
+### Initialization
+
+`ServerCapabilities` may contain a new field to indicate server-side support for this extension:
+
+```typescript
+interface ServerCapabilities {
+  /* ... all fields from the base ServerCapabilities ... */
+
+  /**
+   * The server provides workspace references exporting support.
+   */
+  xworkspaceReferencesProvider?: boolean;
+}
+```
+
+#### Workspace References Request
+
+The workspace references request is sent from the client to the server to locate project-wide references to a symbol given its description / metadata.
+
+_Request_
+* method: 'workspace/xreferences'
+* params: `WorkspaceReferencesParams` defined as follows:
+```typescript
+/**
+ * The parameters of a Workspace References Request.
+ */
+interface WorkspaceReferencesParams {
+    /**
+     * Metadata about the symbol that is being searched for.
```

Comment:
It seems strange to call this a search for a symbol, since it's really about finding references to a symbol that matches a description. Suggested "Description of the symbol for which to find references."